src/utils/response-helper.js

```diff
@@ -7,15 +7,15 @@ function jsonApiError (response) {
   const json = response.json
   let errors = []
 
-  if (Array.isArray(json.errors)) {
+  if (json && Array.isArray(json.errors)) {
     errors = json.errors
   } else {
     // This handles "legacy" errors from our services that haven't changed to
     // JSONAPI-compliant errors yet.
     let errorDetails = []
-    if (json.messages) {
+    if (json && json.messages) {
       errorDetails = json.messages
-    } else if (json.message) {
+    } else if (json && json.message) {
```

Comment:
How does an XHR error response look like? Shouldn't we try to convert it to some useful error?
Right now it just always return with some unhandled error message which I implemented only as a last resort. Doesn't an XHR response contain some useful message that we can extract?

Reply:
![screen shot 2018-11-15 at 13 38 30](https://user-images.githubusercontent.com/17675583/48553429-d727ed00-e8db-11e8-8bf4-7751160e9ccc.png)

It doesn't give many information in this case.

Reply:
That's the `response` argument in this function? Meh 😢